packages/components/pie-button/test/component/pie-button.spec.ts

```diff
@@ -272,422 +156,228 @@ test.describe('Form Actions', () => {
             expect(method).toBe('POST');
         });
 
-        test('should submit the form when pressing Enter with pie-button type `submit`', async ({ page }) => {
-            // Arrange
-            await page.evaluate(() => {
-                const formHTML = `
-                <form id="testForm" action="/foo" method="POST">
-                    <input type="text" id="username" name="username" required>
-                    <input type="password" id="password" name="password" required>
-                    <pie-button id="submitButton" type="submit">Submit</pie-button>
-                </form>
-            `;
-                document.body.innerHTML = formHTML;
-            });
-
-            // Set up the form submission listener
-            await page.evaluate(() => {
-                const form = document.querySelector('#testForm') as HTMLFormElement;
+        const submitTestCases = [
+            { titlePrefix: 'should submit', showSubmitButton: true, expectedFormSubmittedFlagCount: 1 },
+            { titlePrefix: 'should not submit', showSubmitButton: false, expectedFormSubmittedFlagCount: 0 },
+        ];
 
-                form.addEventListener('submit', (e) => {
-                    e.preventDefault();
-
-                    const span = document.createElement('span');
-                    span.id = 'formSubmittedFlag';
-                    document.body.appendChild(span);
-                });
-            });
-
-            // Fill out the form
-            await page.fill('#username', 'testUser');
-            await page.fill('#password', 'testPassword');
-
-            // Act
-            // Press Enter in the password field
-            await page.focus('#password');
-            await page.press('#password', 'Enter');
-
-            // Assert
-            const formSubmittedFlagExists = Boolean(await page.$('#formSubmittedFlag'));
-            expect(formSubmittedFlagExists).toBe(true);
-        });
-
-        test('should NOT submit the form when pressing Enter with pie-button type other than `submit`', async ({ page }) => {
-            // Arrange
-            await page.evaluate(() => {
-                const formHTML = `
-                <form id="testForm" action="/foo" method="POST">
-                    <input type="text" id="username" name="username" required>
-                    <input type="password" id="password" name="password" required>
-                    <pie-button id="resetButton" type="reset">Reset</pie-button>
-                </form>
-            `;
-                document.body.innerHTML = formHTML;
-            });
+        submitTestCases.forEach((testCase) => {
+            test(` ${testCase.titlePrefix} the form when pressing Enter with pie-button type other than 'submit'`, async ({ page }) => {
```

Comment:
not sure if I am reading this wrong.

The original positive test case name was:

`should submit the form when pressing Enter with pie-button type 'submit'`

Now it would be: 

`should submit the form when pressing Enter with pie-button type other than 'submit'`

Is that right?